cql3/statements/permission_altering_statement.cc

```diff
@@ -58,6 +59,11 @@ future<> cql3::statements::permission_altering_statement::check_access(query_pro
                 if (fun->is_native()) {
                     return make_exception_future<>(exceptions::invalid_request_exception("Altering permissions on builtin functions is not supported"));
                 }
+            } else if (v.keyspace()) {
+                // If the resource has a keyspace, it is "all functions in keyspace" and not "all functions"
+                if (is_system_keyspace(*v.keyspace())) {
```

Comment:
Note that this is simple match on the keyspace name, specifically it just checks that the name starts with `system`. This doesn't include all the internal keyspaces like `auth` for example.

Reply:
After reading the error message below I'm no longer sure this does what I think it does. Is this granting access to builting functions?

Reply:
In general, we want to prevent altering permissions on builtin funcitons. This check is preventing altering permissions on functions in the `system` keyspace, because all builtin functions are in this keyspace. The actual test that is failing is https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-daily-release/233/testReport/junit/auth_roles_test/TestAuthRoles/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split002___test_disallow_grant_revoke_on_builtin_functions/
I'll modify the check to explicitly compare the keyspace name to "system". I also just noticed that this check will cover the other case (for single functions), so I'll remove the other if statement for less confusion